### PR TITLE
Rc v0.3.2

### PR DESCRIPTION
**Pull request by rstoneback**
Release Candidate for v0.3.2.

 - Includes fixes to improve building with modern systems
 - Compatible with pysat v3.0+
 - Adds template documents for pulls, community documents, testing via GitHub Actions.
 - Same CDF library version as last release. 

I intend to have a follow up release that increments the CDF library version. For highest backwards compatibility I decided to release this minimum changes version first. 

Note the current version online for PIP only works for Mac systems. This release is intended to bring pysatCDF back from the dead. Rehabilitation and such will follow to further bring pysatCDF up to full project standards. Progress has been made in this release to add template docs, code formatting improved, etc. pysatCDF passes all tests via pysatNASA and is approx. 60 times faster loading CDF files than pure python.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (no docs yet other than readme)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update the version number
- [x] Verify and update Zenodo information in .zenodo.json (Passes linter!, awaiting last confirmation of info)
- [x] doc tests pass locally (pysatCDF doesn't have docs yet. Docs will be added along with multi-dimensional support)
- [x] activate this branch on readthedocs (nothing to activate)
- [x] Upload the test distribution to the testPyPi server (https://test.pypi.org/project/pysatCDF/0.3.2rc1/)
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (I've run integration tests using pysatNASA `optional_pysatcdf` branch. Tests pass locally.)
- [x] Create a wiki page with the release text (https://github.com/pysat/pysat/wiki/pysatCDF-v0.3.2-Release-Text)
- [x] After approval, confirm all changes from the RC pull are listed in the wiki release text.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -190,54 +194,12 @@ def run(self):
     name='pysatCDF.fortran_cdf',
     sources=[os.path.join('pysatCDF', 'fortran_cdf.f')],
     include_dirs=[f2py_cdf_include_path],
-    f2py_options=['--quiet', '--include-paths', f2py_cdf_include_path],  # '--Wall', 'n', '--Wno-tabs', 'n'],
+    f2py_options=['--quiet', '--include-paths', f2py_cdf_include_path],
     extra_objects=[f2py_cdf_lib_path],
+    extra_f77_compile_args=['--std=legacy'],
```
Comment (rstoneback) [inline setup.py:199]: This was a key input to get things going again.

**Comment by rstoneback** [comment]
@gregstarr @asreimer Thanks for contributing to pysatCDF! Apologies it has taken so long to get this release out. I've updated the zenodo.json with my best attempt at correct author information. I'd appreciate it if you could review the info.

**Comment by gregstarr** [comment]
> @gregstarr @asreimer Thanks for contributing to pysatCDF! Apologies it has taken so long to get this release out. I've updated the zenodo.json with my best attempt at correct author information. I'd appreciate it if you could review the info.

Looks good to me, thanks!

**Review by jklenzing — CHANGES_REQUESTED** [review]
It's looking good.  Some of the docstrings are missing, but we could bump that to the next version.  Running flake8 locally (with the extra docstring checks), I see 53 issues.  Most can be pushed off until the doc update, I've flagged some of the more critical ones.

/pysatCDF/_cdf.py:11:6: H304: No relative imports. 'from . import fortran_cdf' is a relative import

File: setup.py [hunk 2]
```diff
@@ -14,66 +12,68 @@
 from numpy.distutils.command.build import build
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply setup.py:12]: can't flag the location, but referring to `numpy.distutils.core`

**Comment by jklenzing** [comment]
Running unit tests for https://github.com/pysat/pysatNASA/pull/102 with this branch, I get the following error when trying to load any cdf instrument:
```
>       meta = pysat.Meta(pandas.DataFrame.from_dict(self.meta, orient='index'),
                          units_label=units_label, name_label=name_label,
                          fill_label=fill_label, plot_label=plot_label,
                          min_label=min_label, max_label=max_label,
                          notes_label=notes_label, desc_label=desc_label,
                          axis_label=axis_label)
E       TypeError: __init__() got an unexpected keyword argument 'units_label'
```
it stems from the `to_pysat` function.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  are welcome :)
+
+Development
+-----------
+
+To set up `pysatCDF` for local development:
+
+1. [Fork pysat on GitHub](https://github.com/pysat/pysatCDF/fork).
+
+3. Clone your fork locally:
```
Comment (jklenzing) [inline CONTRIBUTING.md:49]:
```suggestion
2. Clone your fork locally:
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+   and Coveralls)
+
+If you don't have all the necessary Python versions available locally or
+have trouble building all the testing environments, you can rely on
+the project's Continuous Integration (CI) service to run the tests for each change you add in the pull
+request. Because testing here will delay tests by other developers,
+please ensure that the code passes all tests on your local system first.
+
+Project Style Guidelines
+^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (jklenzing) [inline CONTRIBUTING.md:114]: This isn't formatting correctly for some reason.

File: pysatCDF/_cdf.py [hunk 5]
```diff
@@ -271,45 +262,50 @@ def _call_multi_fortran_z(self, names, data_types, rec_nums,
         Parameters
         ----------
         names : list_like
```
Comment (aburrell) [inline pysatCDF/_cdf.py:264]:
```suggestion
        names : list-like
```

Here and elsewhere.

File: pysatCDF/_cdf.py [hunk 6]
```diff
@@ -320,16 +316,19 @@ def _call_multi_fortran_z(self, names, data_types, rec_nums,
...
                 self._process_return_multi_z(data, sub_names, sub_sizes)
             else:
                 raise IOError(fortran_cdf.statusreporter(status))
 
+        return
+
     def _process_return_multi_z(self, data, names, dim_sizes):
-        """process and attach data from fortran_cdf.get_multi_*"""
-        # process data
+        """Process and attach data from fortran_cdf.get_multi_*"""
```
Comment (aburrell) [inline pysatCDF/_cdf.py:330]:
```suggestion
        """Process and attach data from various `fortran_cdf` 'get' functions."""
```

File: pysatCDF/_cdf.py [hunk 7]
```diff
@@ -518,43 +540,42 @@ def _call_multi_fortran_z_attr(self, names, data_types, num_elems,
                 self._process_return_multi_z_attr(data, sub_names,
                                                   sub_var_names, sub_num_elems)
             else:
-                # raise ValueError('CDF Error code :', status)
+                # Raise first error.
```
Comment (aburrell) [inline pysatCDF/_cdf.py:543]:
```suggestion
                # Raise the first error.
```

File: setup.cfg [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+max-line-length = 80
+ignore =
+  D200
+  D202
+  W503
+  pysatCDF/__init__.py F401
+
+exclude = conf.py
+  tests
+  __init__.py
```
Comment (aburrell) [inline setup.cfg:58]:
```suggestion
```

File: setup.py [hunk 9]
```diff
@@ -2,78 +2,78 @@
...
+# pysatCDF specific CDF install
 base_cdf = None
 build_cdf_flag = True
 
-# leave items below to None
+# Leave items below to None
 # name of library, e.g. for mac os x, libcdf.a
 lib_name = None
-# shared library name, needed for some systems ( do not use Mac OS X)
+# Shared library name, needed for some systems (do not use on Mac OS X)
```
Comment (aburrell) [inline setup.py:26]:
```suggestion

# Shared library name, needed for some systems (do not use on Mac OS X)
```

File: setup.py [hunk 10]
```diff
@@ -113,131 +110,89 @@ def CDF_build(self, ppath):
...
         self.extensions[0].include_dirs = [os.path.join(lib_path, 'include')]
-        self.extensions[0].f2py_options = ['--include-paths', os.path.join(lib_path, 'include'), '--quiet']
-        self.extensions[0].extra_objects = [os.path.join(lib_path, 'lib', lib_name)]
-        # add shared library, if provided
+        self.extensions[0].f2py_options = ['--include-paths',
+                                           os.path.join(lib_path, 'include'),
+                                           '--quiet']
+        self.extensions[0].extra_objects = [os.path.join(lib_path, 'lib',
+                                                         lib_name)]
+        # Add shared library, if provided
```
Comment (aburrell) [inline setup.py:161]:
```suggestion

        # Add shared library, if provided
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
+---
+name: Bug report
+about: Create a report to help us improve
+title: ''
+labels: ''
+assignees: ''
+
+---
+
+**Describe the bug**
```
[the remark this answers is not in the document]
Comment (rstoneback) [reply .github/ISSUE_TEMPLATE/bug_report.md:10]: Thanks for the note. I copied the templates from somewhere in our ecosystem. I don't remember from which package now though. Will update.

[the remark this answers is not in the document]
Comment (rstoneback) [reply .github/ISSUE_TEMPLATE/bug_report.md]: pysatMadrigal and pysatSpaceWeather have 
```
# Description
A clear and concise description of what the bug is, including a description
of what you expected the outcome to be.

```

while the others, pysat, Models, Missions, Seasons, and CDAAC have the same here. I'll copy from Madrigal/SpaceWeather. Went a bit out of strict ** to # replacement, but the SpaceWeather/Madrigal versions are two whole years newer than the other packages.

**Comment by rstoneback** [comment]
Thanks for the feedback!

**Review by jklenzing — APPROVED** [review]
Looks good!

**Comment by rstoneback** [comment]
Thanks everyone!

**Pull request merged**